### ISSUE TITLE
How to change batch size?

### ISSUE THREAD

**Issue by Hubotcoder** [xnyuwg/procnet#1]
Hi! Thank you for your work. It's really easy to deploy this project.
I have a problem.
I change the default batch_size from 32 to 256, in both run.sh and run.py, but it seems didn't work at all.
How can I change the batch size?

**Comment by xnyuwg** [comment]
Thank you for your interest.

Sorry, it may be confusing in the code. The term "batch size" in the run.sh script refers to the logical batch size, not the physical batch size that the GPU processes. In our implementation, we employ gradient accumulation, and the GPU always processes a physical batch size of 1. Therefore, when the "batch size" in run.sh is set to 32, it essentially means the logical batch size is 32 (gradient accumulation step * 1 physical batch size in GPU = 32). 

Adjusting the "batch size" in run.sh essentially modifies the gradient accumulation step. While this doesn't alter the training time for a single epoch, it may influence the performance and the time it takes to converge.

The reason our code is structured this way is primarily for ease of initial debugging. The forward function processes just one document in each iteration. Given that processing a single document consumes about 10G of GPU RAM and our GPU has a total of 24G, even if we were to modify the code to support larger physical batch sizes, our GPU could only handle a batch size of 2. Given this limited improvement and the benefits of easier debugging and improved code readability, we chose to maintain the code limited to a physical batch size of 1.

**Comment by xnyuwg** [comment]
Thank you for your interest.

Sorry, it may be confusing in the code. The term "batch size" in the run.sh script refers to the logical batch size, not the physical batch size that the GPU processes. In our implementation, I employ gradient accumulation, and the GPU always processes a physical batch size of 1. Therefore, when the "batch size" in run.sh is set to 32, it essentially means the logical batch size is 32 (gradient accumulation step * 1 physical batch size in GPU = 32).

Adjusting the "batch size" in run.sh essentially modifies the gradient accumulation step. While this doesn't alter the training time for a single epoch, it may influence the performance and the time it takes to converge.

The reason our code is structured this way is primarily for ease of initial debugging. The forward function processes just one document in each iteration. Given that processing a single document consumes about 10G of GPU RAM and our GPU has a total of 24G, even if I was to modify the code to support larger physical batch sizes, our GPU could only handle a batch size of 2. Given this limited improvement and the benefits of easier debugging and improved code readability, I chose to maintain the code limited to a physical batch size of 1.

**Comment by Hubotcoder** [comment]
> 

Thank you for your explanation! It is really clear now.

**Comment by Hubotcoder** [comment]
> Thank you for your interest.
> 
> Sorry, it may be confusing in the code. The term "batch size" in the run.sh script refers to the logical batch size, not the physical batch size that the GPU processes. In our implementation, I employ gradient accumulation, and the GPU always processes a physical batch size of 1. Therefore, when the "batch size" in run.sh is set to 32, it essentially means the logical batch size is 32 (gradient accumulation step * 1 physical batch size in GPU = 32).
> 
> Adjusting the "batch size" in run.sh essentially modifies the gradient accumulation step. While this doesn't alter the training time for a single epoch, it may influence the performance and the time it takes to converge.
> 
> The reason our code is structured this way is primarily for ease of initial debugging. The forward function processes just one document in each iteration. Given that processing a single document consumes about 10G of GPU RAM and our GPU has a total of 24G, even if I was to modify the code to support larger physical batch sizes, our GPU could only handle a batch size of 2. Given this limited improvement and the benefits of easier debugging and improved code readability, I chose to maintain the code limited to a physical batch size of 1.
